src/Slawek/Fsd.Slawek.Cs.Ex2/Fsd.Slawek.Cs.Ex2/Models/AnimalsGenerator.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Fsd.Slawek.Cs.Ex2.Models
+{
+    public class AnimalsGenerator
+    {
+        string[] kindsOfAnimals = new string[] { "cat", "dog", "horse", "cow" };
```

Comment:
a może zamiast przechowywania w tablicy stringów jako kluczy które wskazują na słownik mapujący string z typem klasy przechowasz sobie listę typów? dzięki temu nie będziesz musiał miec fabryki. Ale oczywiście to co zrobiłes jest bardzo spoko, to o czym ja wspominam to trochę wyższy level. Coś w stylu x = new List<Type> { typeof(Cat) }. Żeby stworzyć obiekt klasy musisz użyć czegoś takiego jak Activator.CreateInstance. Możesz zostawić lub się pobawić - twoja decyzja